### ISSUE TITLE
MyMultiheadAttention 中的 padding 操作

### ISSUE THREAD

**Issue by linling1** [moon-hotel/TransformerTranslation#2]
`if key_padding_mask is not None:
        attn_output_weights = attn_output_weights.view(bsz, num_heads, tgt_len, src_len)
        # 变成 [batch_size, num_heads, tgt_len, src_len]的形状
        attn_output_weights = attn_output_weights.masked_fill(
            key_padding_mask.unsqueeze(1).unsqueeze(2),  # 扩展维度，从[batch_size,src_len]变成[batch_size,1,1,src_len]
            float('-inf'))  #
        attn_output_weights = attn_output_weights.view(bsz * num_heads, tgt_len,
                                                       src_len)  # [batch_size * num_heads, tgt_len, src_len]`

中的 `attn_output_weights = attn_output_weights.masked_fill(padding_mask.unsqueeze(1).unsqueeze(2), float('-inf'))` 是否应该是 `attn_output_weights = attn_output_weights.masked_fill(padding_mask.unsqueeze(1).unsqueeze(2) == False, float('-inf'))`  ？？

**Comment by moon-hotel** [comment]
首先，masked_fill()函数的第一个参数指定的是需要被mask的位置，在这里会padding传入值为TRUE的位置。由于padding_mask = (src == self.PAD_IDX).transpose(0, 1)是这样产生的，即需要mask的地方已经被标记为TRUE了，所以src中需要被padding的地方在padding_mask中就已经是TRUE了，所以直接传入padding_mask即可。 当然，如果padding_mask = (src != self.PAD_IDX)是这样产生的，那么就会得到一个类似于[True,True,True,False,False]的向量，此时需要对最后两个位置进行padding，所以就需要以这样的方式传入masked_fill(padding_mask==False)

**Comment by linling1** [comment]
那为啥你blog里的示例 padding_mask 都是用 False 表示需要被 mask 的位置了？
`src_key_padding_mask = torch.tensor([[True, True, True, False, False],
                                                                      [True, True, True, True, False]])`

**Comment by moon-hotel** [comment]
用False表示需要被mask的位置没错，而masked_fill方法的在padding时只会对True上的位置进行padding，所以如果key_padding_mask = [True, True,True, False,False]后面两个FALSE需要padding，那么自然是masked_fill(key_padding_mask==False)这样来使用。
